src/coreclr/vm/classlayoutinfo.cpp

```diff
@@ -653,8 +653,9 @@ VOID EEClassLayoutInfo::CollectLayoutFieldMetadataThrowing(
         DEBUGARG(szName)
         );
 
-    // Type is blittable only if parent is also blittable
-    isBlittable = isBlittable && (fHasNonTrivialParent ? pParentMT->IsBlittable() : TRUE);
+    // Type is blittable only if parent is also blittable and is not empty.
+    isBlittable = isBlittable && (fHasNonTrivialParent
```

Comment:
This is a bit dense. May I suggest:

```
if (isBlittable && fHasNonTrivialParent)
{
    isBlittable = pParentMT->IsBlittable()  // Check parent
        && (!pParentLayoutInfo || !pParentLayoutInfo->IsZeroSized()); // Ensure non-zero size
}
```

The logic seems fine - the current style is less than appealing.